### PR TITLE
feat: get all signed users chat

### PR DESCRIPTION
**Pull request by falhar**

### CODE REVIEW

**Review by n-permana — APPROVED** [review]

**Review by brm-stnd — APPROVED** [review]

File: __tests__/api/chat/channels.test.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      .set({Authorization: `Bearer invalid-token`})
+      .expect(401);
+    expect(response.body).toEqual(
+      expect.objectContaining({
+        code: 401,
+        message: expect.any(String),
+        data: null
+      })
+    );
+  });
```
Comment (therour) [inline __tests__/api/chat/channels.test.js:61]: untuk testing 401 bisa pake`createReusableAuthTestSuite` buatan mas Fajar

```js
const {createReusableAuthTestSuite} = require('../../__authTest__/createReusableAuthTestSuite');

createReusableAuthTestSuite(supertest(app).get('/api/v1/chat/channels/signed'));
```

File: services/chat/__mocks__/stream-chat.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  connectUser = jest.fn();
+  disconnectUser = jest.fn();
+  // Mock the queryChannels method
+  queryChannels = jest.fn(async () => {
+    return [{data: {}, config: {}, own_capabilities: {}, state: {members: [{}], messages: ''}}];
+  });
+}
+
+// Export the mocked StreamChat class
+module.exports = {StreamChat: MockStreamChat};
```
Comment (therour) [inline services/chat/__mocks__/stream-chat.js:18]: @falhar  file mock ini bisa dijadiin global mock, tinggal ditaruh di root aja `__mocks__/stream-chat.js`

File: __tests__/api/chat/channels.test.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    expect(StreamChat.getInstance).toHaveBeenCalled();
+    expect(StreamChat._instance).toBeInstanceOf(StreamChat);
+    expect(StreamChat._instance.connectUser).toHaveBeenCalledWith(
+      {id: user.user_id},
+      'valid-token'
+    );
+    expect(StreamChat._instance.queryChannels).toHaveBeenCalledWith(
+      {type: 'messaging', members: {$in: [user.user_id]}},
+      [{last_message_at: -1}]
+    );
```
Comment (therour) [inline __tests__/api/chat/channels.test.js:45]: nice Testing for stream-chat 👍

**Comment by therour** [comment]
Nice coverage, at least `chatController.getChannels` is already covered by testing 👍

**Pull request merged**